spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/ReplyingMessageListener.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.listener.adapter;
+
+/**
+ * A Message Listener that returns a reply - intended for lambda use in a
+ * {@link MessageListenerAdapter}.
+ *
+ * @author Gary Russell
+ * @since 2.0
+ *
+ */
+@FunctionalInterface
+public interface ReplyingMessageListener<T, R> {
+
+	/**
+	 * Handle the message and return a reply.
+	 * @param t the request.
+	 * @return the reply.
+	 */
+	R handleMessage(T t);
```

Comment:
Ah! Sure! `handleMessage` default method does the trick :smile: 

Although it is really dilemma, because we have many similar interfaces which could be replaced by the `java.util.function` stuff...
Not sure that SF has removed some of their in favor of those generics, but right, we should still keep some backward compatibility, if case of implementations. 
So, yeah, let's leave it as is for now!

I think the PR is cool and I'll merge it over a time.

Busy with `ScopedProxyFactoryBean` hell :angry: